File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        
        Set<Integer> lostSet = new HashSet<>();
        Set<Integer> reserveSet = new HashSet<>();
        
        for (int l : lost) lostSet.add(l);
        for (int r : reserve) reserveSet.add(r);
        
        Set<Integer> commonSet = new HashSet<>(lostSet);
        commonSet.retainAll(reserveSet);
        lostSet.removeAll(commonSet);
        reserveSet.removeAll(commonSet);
        
        List<Integer> sortedReserve = new ArrayList<>(reserveSet);
        Collections.sort(sortedReserve);
        
        for (int r : sortedReserve) {
            if (lostSet.contains(r - 1)) {
                lostSet.remove(r - 1);
            } else if (lostSet.contains(r + 1)) {
                lostSet.remove(r + 1);
            }
        }
        
        int answer = n - lostSet.size();
        
        return answer;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = n
    stds = [0] * n
    
    for lo in lost:
        stds[lo-1] -= 1
    
    for re in reserve:
        stds[re-1] += 1
    
    for i in range(n):
        if stds[i] == -1:
            if i > 0 and stds[i-1] == 1:
                stds[i-1] -= 1
                stds[i] += 1
            elif i < n-1 and stds[i+1] == 1:
                stds[i] += 1
                stds[i+1] -= 1
            else:
                answer -= 1

    return answer